### PR TITLE
Added option to define config.js for st2web

### PR DESCRIPTION
**Pull request by moonrail**
Hello maintainers,

while testing and developing internally we've experienced the requirement, where we would have to access StackStorm in a subpath of a FQDN, e.g. `sub.domain.tld/stackstorm`.

For this configuration is required for Ingress and `st2web`.
<br>This PR adds the possibility to define configuration for `st2web`.

Ingress is pretty straight forward and already supported, e.g.:
```yaml
ingress:
    enabled: true
    annotations:
      kubernetes.io/ingress.class: nginx
      kubernetes.io/tls-acme: 'true'
      nginx.ingress.kubernetes.io/rewrite-target: /$1
    hosts: 
      - host: 'sub.domain.tld'
        paths:
          - path: '/stackstorm/(.*)'
            serviceName: st2web
            servicePort: 80
    tls:
      - secretName: 'tls-sub.domain.tld'
        hosts:
          - 'sub.domain.tld'
```

With this Ingress configuration, `st2web` is properly reachable.

By default `st2web` uses relative paths to its domain to access services like `auth`, `api`, and `stream`. So for our example, `st2web` would try to authenticate against `sub.domain.tld/auth`.
<br>A solution is described here: https://github.com/StackStorm/st2web#connecting-to-st2-server

In short: Edit /opt/stackstorm/static/webui/config.js to include complete Endpoints for `auth`, `api`, and `stream`.

For our example config.js could be:
```yaml
'use strict';

/* global angular */
angular.module('main')
  .constant('st2Config', {

    hosts: [
      {
        name: 'test',
        url: 'https://sub.domain.tld/stackstorm/api',
        auth: 'https://sub.domain.tld/stackstorm/auth',
        stream: 'https://sub.domain.tld/stackstorm/stream',
      }
    ]

  });
```

### CODE REVIEW

File: templates/deployments.yaml [hunk 1]
```diff
@@ -409,13 +409,27 @@ spec:
...
         resources:
 {{ toYaml .Values.st2web.resources | indent 10 }}
     {{- if .Values.st2web.serviceAccount.attach }}
       serviceAccountName: {{ template "stackstorm-ha.serviceAccountName" . }}
     {{- end }}
+    {{- if .Values.st2web.config }}
+      volumes:
+        - name: st2web-config-vol
+          configMap:
+            name: {{ .Release.Name }}-st2web-config
```
Comment (armab) [inline templates/deployments.yaml:429]: To make sure the st2web deployment redeploys the pods on config change, we'll need to include the Configmap checksum via  `annotations`.

Take a look at other examples like 
https://github.com/StackStorm/stackstorm-ha/blob/1c5103880f2e80f9ae5eb1d97c218aa7717a3258/templates/deployments.yaml#L33-L35

**Review by armab — CHANGES_REQUESTED** [review]
Thanks for the contribution! That's a nice composed feature :+1: 

I just left one comment before we can merge it.

**Comment by armab** [comment]
Please also sync up your PR branch with the latest `master`, - that'll fix the helm-lint failures.

**Comment by moonrail** [comment]
Rebased :)

**Comment by armab** [comment]
Thanks @moonrail!
The only missing thing to merge it is checksum adjustment for the new configmap/deployment, see the review comment above ^^: https://github.com/StackStorm/stackstorm-ha/pull/165#discussion_r559191746

File: templates/deployments.yaml (hunk 1 above)
Comment (moonrail) [reply templates/deployments.yaml:431]: done :)

**Comment by moonrail** [comment]
@armab 
Sorry, I've been busy and overlooked your comment - Checksum should be generated as with other deployments as of now.

**Comment by armab** [comment]
You're definitely faster then I am with the other PRs :)

**Review by armab — APPROVED** [review]
Nice feature, thanks a lot for implementation! :+1:
